contracts/HubPool.sol

```diff
@@ -29,16 +31,14 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         bool isEnabled;
     }
 
-    enum RefundRequestStatus {
-        Pending,
-        Finalized,
-        Disputed
-    }
-
     struct RelayerRefundRequest {
-        bytes32 poolRebalanceProof;
-        bytes32 destinationDistributionProof;
-        RefundRequestStatus status;
+        uint64 requestExpirationTimestamp;
+        uint64 unclaimedPoolRebalanceLeafs;
+        bytes32 poolRebalanceRoot;
+        bytes32 destinationDistributionRoot;
+        mapping(uint256 => uint256) claimedBitMap;
+        address proposer;
+        bool proposerBondRepaid;
     }
 
     RelayerRefundRequest[] public relayerRefundRequests;
```

Comment:
Do you think it's possible to somehow avoid storing all of these since only one is active at a time? We could reuse the same storage slot each time and always know if the prev one has been closed out when starting the next one because the storage slot isn't zero'd out. Note: only problem I see with this approach is that it might not work well with the mapping, since you can't really close out/delete a mapping.

Reply:
we can do that. your goal is we'll get a refund if we re-use the same storage slots between refunds, right?

Reply:
one option would to simply store one instance of `RelayerRefundRequest` within the hubPool. it tracks the exact variables we have in it now, but it's not within an array. When the `unclaimedPoolRebalanceLeafs` hits 0 and a data worker calls `initiateRelayerRefund` again we simply delete this instance of this struct and re-set it. this way we only ever have the storage of _one_ instance of this data and will net many of the refunds. I'll give this a go and see how it looks.

Reply:
I tried this out and it does not really work as you cant delete/reset the mapping in the way we'd need to to work. I tried creating this basic test:
```
pragma solidity ^0.8.0;

contract StructTest {
    struct RelayerRefundRequest {
        uint64 requestExpirationTimestamp;
        uint64 unclaimedPoolRebalanceLeafs;
        bytes32 poolRebalanceRoot;
        bytes32 destinationDistributionRoot;
        mapping(uint256 => uint256) claimedBitMap;
        address proposer;
        bool proposerBondRepaid;
    }

    RelayerRefundRequest public activeRelayerRefundRequest;

    function setValue() public {
        activeRelayerRefundRequest.proposer = msg.sender;
        activeRelayerRefundRequest.claimedBitMap[0] = 10;
    }

    function getClaim(uint256 index) public view returns (uint256) {
        return activeRelayerRefundRequest.claimedBitMap[index];
    }

    function resetStruct() public {
        delete activeRelayerRefundRequest;
    }
}
```

The delete call worked fine at resetting the `proposer` but after deleting the instance of `RelayerRefundRequest` the `claimedBitMap` was still initialised on the value set in the `setValue` call.

Reply:
long story short you cant use `delete` on complex data types and so cant reset it. this means we either need:
1) to create a 3d mapping where we index some information to spectate out `claimedBitMap` for each proposal 2) do an array like I originally proposed 3) move away from a claimedBitMap to some other structure for tracking claims. I personally think leaving it like it is now is probs the cleanest and cheapest option.

Reply:
@chrismaree as discussed IRL, I think we landed on making the bitmap a single integer and limit us to supporting 256 chains.

Reply:
That's right. I've implemented this in a separate PR to try minimise the changes here: https://github.com/across-protocol/across-smart-contracts-v2/pull/14